### ISSUE TITLE
[BUG] Chapter 7: Wrong data indexed in Random Forests section, cell 21

### ISSUE THREAD

**Issue by hjstein** [ageron/handson-ml2#474]
**Describe the bug**
In notebook https://github.com/ageron/handson-ml2/blob/master/07_ensemble_learning_and_random_forests.ipynb, cell 21 (the beginning of Section 4 on random forests), you randomly select indices from `range(0, len(X_train))`, but use those indices to index into `X`.  This if `X` started with the training data & was followed by the test data, then this would effectively be reordering `X_train`, but it's not because `X_train` is generated by `test_train_split()`, so this doesn't look right.

This section doesn't appear in the book, so I don't have any more details as to what it is intended to show, but it appears to be demonstrating the sensitivity of the decision tree classifier to the order of the data.

So, to be consistent with the other analyses, I believe that:
```python
    tree_clf.fit(X[indices_with_replacement], y[indices_with_replacement])
```
should instead be:
```python
    tree_clf.fit(X_train[indices_with_replacement], y_train[indices_with_replacement])
```

Or, if the intent is to show the dependency on the test set selection, then the code should instead select `len(X_train)` samples from `X`, as in:
```python
    indices_with_replacement = np.random.randint(0, len(X), len(X_train))
    tree_clf.fit(X[indices_with_replacement], y[indices_with_replacement])
```

**To Reproduce**

Analysis and issue noted above.

**Expected behavior**

See above.

**Screenshots**
The original code and the two above modifications:

![Screenshot 2021-10-03 at 17-54-24 07_ensemble_learning_and_random_forests - Jupyter Notebook](https://user-images.githubusercontent.com/7682665/135772731-aeae87fd-8cba-43ae-8e5b-68aff3e5dd38.png)

**Versions (please complete the following information):**
 - OS: Linux
 - Python: 3.7.4
 - TensorFlow: Unused
 - Scikit-Learn: 1.0
- Other libraries that may be connected with the issue:
   - None that I'm aware of

**Additional context**
None

**Comment by ageron** [comment]
Hi  @hjstein ,
Great catch, thanks a lot for the detailed analysis!

Indeed, you're right, I should have written `X_train` and `y_train` instead of `X` and `y`:

```python
    tree_clf.fit(X_train[indices_with_replacement], y_train[indices_with_replacement])
```

Sorry about that, I'll fix this now.
